src/aro/Value.zig

```diff
@@ -927,6 +1003,10 @@ pub fn compare(lhs: Value, op: std.math.CompareOperator, rhs: Value, comp: *cons
     return lhs_bigint.order(rhs_bigint).compare(op);
 }
 
+pub fn compare(lhs: Value, op: std.math.CompareOperator, rhs: Value, comp: *const Compilation) bool {
+    return lhs.compareExtra(op, rhs, comp).?;
```

Comment:
This looks dangerous. Is it expected that other places won't reach this or is handling it just out of scope for this PR? Shouldn't be a blocker but a comment reminder wouldn't hurt.

Reply:
I don't think other paths could reach that, but I changed it to use a separate function for comparing pointers. In doing so I discovered a silly oversight with subtraction where I'd switched the operand order :)